main/views.py

```diff
@@ -176,25 +179,30 @@ def logout_view(request):
     return redirect('/')
 
 
-class CustomUserCreationForm(UserCreationForm):
-    class Meta(UserCreationForm.Meta):
-        model = User
-        fields = ('email','first_name','last_name','space',)
-
-
 class SignupView(CreateView):
     form_class = CustomUserCreationForm
     model = User
     template_name = 'main/signup.html'
-    success_url = "/home"
 
     def form_valid(self, form):
-        res = super().form_valid(form)
-        user = authenticate(username=form.cleaned_data['email'], password=form.cleaned_data['password1'])
-        if user is not None:
-            if user.is_active:
-                login(self.request, user)
-        return res
+        obj = form.save(commit=False)
+        obj.set_password(User.objects.make_random_password())
+        obj.save()
+
+        # PasswordResetForm only requires the "email" field, so will validate.
+        reset_form = PasswordResetForm(self.request.POST)
+        reset_form.is_valid()  # Must trigger validation
+        # Copied from django/contrib/auth/views.py : password_reset
+        opts = {
+            'use_https': self.request.is_secure(),
+            'email_template_name': 'main/verification.html',
+            'subject_template_name': 'main/verification_subject.txt',
+            'request': self.request,
+        }
+        # This form sends the email on save()
+        reset_form.save(**opts)
+
+        return redirect('signup-done')
```

Comment:
deally this piece of code should first send the email, then save the data - otherwise issues with email sending will cause us to have a saved user that nobody has links for (discovered this by failing to configure my smtp ;)